src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementShortlinkEntityUpdater.java

```diff
@@ -0,0 +1,120 @@
+package fi.otavanopisto.kuntaapi.server.integrations.management;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import javax.annotation.Resource;
+import javax.ejb.AccessTimeout;
+import javax.ejb.Singleton;
+import javax.ejb.TimerService;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import fi.metatavu.management.client.ApiResponse;
+import fi.metatavu.management.client.DefaultApi;
+import fi.metatavu.management.client.model.Shortlink;
+import fi.otavanopisto.kuntaapi.server.cache.ModificationHashCache;
+import fi.otavanopisto.kuntaapi.server.controllers.IdentifierController;
+import fi.otavanopisto.kuntaapi.server.controllers.IdentifierRelationController;
+import fi.otavanopisto.kuntaapi.server.discover.EntityUpdater;
+import fi.otavanopisto.kuntaapi.server.id.ShortlinkId;
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.otavanopisto.kuntaapi.server.integrations.management.cache.ManagementShortlinkCache;
+import fi.otavanopisto.kuntaapi.server.integrations.management.tasks.ShortlinkIdTaskQueue;
+import fi.otavanopisto.kuntaapi.server.persistence.model.Identifier;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask.Operation;
+
+@ApplicationScoped
+@Singleton
+@AccessTimeout (unit = TimeUnit.HOURS, value = 1l)
+@SuppressWarnings ("squid:S3306")
+public class ManagementShortlinkEntityUpdater extends EntityUpdater {
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private ManagementTranslator managementTranslator;
+  
+  @Inject
+  private ManagementApi managementApi;
+  
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private IdentifierRelationController identifierRelationController;
+
+  @Inject
+  private ManagementShortlinkCache managementShortlinkCache;
+  
+  @Inject
+  private ModificationHashCache modificationHashCache;
+
+  @Inject
+  private ShortlinkIdTaskQueue shortlinkIdTaskQueue;
+  
+  @Resource
+  private TimerService timerService;
+
+  @Override
+  public String getName() {
+    return "management-shortlinks";
+  }
+
+  @Override
+  public void timeout() {
+    executeNextTask();
+  }
+
+  private void executeNextTask() {
+    IdTask<ShortlinkId> task = shortlinkIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateManagementShortlink(task.getId(), task.getOrderIndex()); 
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteManagementShortlink(task.getId());
+      }
+    }
+  }
+  
+  private void updateManagementShortlink(ShortlinkId shortlinkId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateManagementShortlink" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)